inspire_matcher/core.py

```diff
@@ -192,42 +192,45 @@ def _compile_nested(query, record):
             },
         },
     }
+    return nested_query, paths, search_paths
+
 
+def _compile_nested(query, record):
+    nested_query, paths, search_paths = _create_nested_template(query)
     for path, search_path in zip(paths, search_paths):
         value = get_value(record, path)
         if not value:
             return
 
-        result['query']['nested']['query']['bool']['must'].append({
+        nested_query['query']['nested']['query']['bool']['must'].append({
             'match': {
                 search_path: value,
             },
         })
 
-    return result
+    return nested_query
 
 
 def _compile_nested_prefix(query, record):
-    path, search_path = query['path'], query['search_path']
-    value = get_value(record, path)
-    if not value:
-        return
-
-    nested_path = search_path.split('.')[0]
-    result = {
-        "query": {
-            "nested": {
-                "path": nested_path,
-                "query": {
-                    'match_phrase_prefix': {
-                        search_path: value,
-                    }
+    nested_query, paths, search_paths = _create_nested_template(query)
+    for path, search_path in zip(paths, search_paths):
+        value = get_value(record, path)
+        if not value:
+            return
+        if "journal_title" in path:
```

Comment:
I wouldn't bind it to a specific field, inspire-matcher doesn't have and shouldn't have any knowledge about the inspirehep model, it's used both for workflows and for records. We could add more info in the config to know which field to choose.